contrib/cirrus/lib.sh

```diff
@@ -288,3 +288,40 @@ remove_packaged_podman_files() {
     # Be super extra sure and careful vs performant and completely safe
     sync && echo 3 > /proc/sys/vm/drop_caches || true
 }
+
+# Execute make localbenchmarks in $CIRRUS_WORKING_DIR/data
+# for preserving as a task artifact.
+localbenchmarks() {
+    local datadir
+    req_env_vars DISTRO_NV PODBIN_NAME PRIV_NAME TEST_ENVIRON TEST_FLAVOR
+    req_env_vars VM_IMAGE_NAME EC2_INST_TYPE
+
+    datadir=$CIRRUS_WORKING_DIR/data
+    mkdir -p $datadir
+
+    # Most of these are defined by cirrus and it's YAML.
+    # shellcheck disable=SC2154,SC2153
+    cat <<EOF>$datadir/benchmarks.env
+CIRRUS_BASE_BRANCH=$CIRRUS_BASE_BRANCH
+CIRRUS_BRANCH=$CIRRUS_BRANCH
+CIRRUS_BUILD_ID=$CIRRUS_BUILD_ID
+CIRRUS_CHANGE_IN_REPO=$CIRRUS_CHANGE_IN_REPO
+CIRRUS_CRON=$CIRRUS_CRON
+CIRRUS_PR=$CIRRUS_PR
+CIRRUS_TASK_ID=$CIRRUS_TASK_ID
+DISTRO_NV=$DISTRO_NV
+PODBIN_NAME=$PODBIN_NAME
+PRIV_NAME=$PRIV_NAME
+TEST_ENVIRON=$TEST_ENVIRON
+TEST_FLAVOR=$TEST_FLAVOR
+VM_IMAGE_NAME=$VM_IMAGE_NAME
+INST_TYPE=$EC2_INST_TYPE
```

Comment:
You know how I feel about duplication. (Prime example right here, where one envariable is misleadingly called the wrong name). Since `$SECRET_ENV_RE` is so nicely defined above, could you instead do
```bash
    env | grep -Ev "$SECRET_ENV_RE" | sort > $datadir/benchmarks.env
```

Reply:
> one envariable is misleadingly called the wrong name

`INST_TYPE` was done on purpose since it's not the consumers job to pick apart cloud-differences.  The only reason we don't use an `INST_TYPE` globally is Cirrus-CI didn't support it initially.  Now it does, and *someone* (me) might decide to go fix it one day :wink: 

I can do the sorting and filtering though, that's a no-brainier.